app/services/spotify_service.rb

```diff
@@ -1,28 +1,88 @@
 class SpotifyService
-  def self.get(url)
-    response = conn.get(url)
+  def initialize(refresh_token)
+    @access_token = get_access_token(refresh_token)
```

Comment:
I really like this since it will make sure every time a user goes to a page, their access token will still be active.